app/services/articles/destroyer.rb

```diff
@@ -3,12 +3,14 @@ module Destroyer
     module_function
 
     def call(article, event_dispatcher = Webhook::DispatchEvent)
-      article.destroy!
       Notification.remove_all_without_delay(notifiable_ids: article.id, notifiable_type: "Article")
+
       if article.comments.exists?
-        Notification.remove_all(notifiable_ids: article.comments.pluck(:id),
-                                notifiable_type: "Comment")
+        Notification.remove_all(notifiable_ids: article.comments.pluck(:id), notifiable_type: "Comment")
       end
+
+      article.destroy!
```

Comment:
moved this as the last step mainly for a reason: related objects should be removed before the object itself is deleted